src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  userId: number,
+  selectedTodo: Todo | null,
+  clearSelectedTodo: () => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  userId,
+  selectedTodo,
+  clearSelectedTodo,
+}) => {
+  const [user, setUser] = useState<null | User >(null);
+
+  useEffect(() => {
+    getUser(userId)
+      .then(setUser)
+      .catch(error => (new Error(error.message)));
+  }, [user]);
```

Comment:
Since you set the value of the `user` in the middle of the `useEffect` hook via `setUser`, the reference to the `user` object as a dependency for `useEffect` is not correct, and in this case it would be more correct to refer to the `userId` or not to transfer anything to the dependency array at all, because we need to get the data for a component only once.